src/main/java/ser/JsonEncoder.java

```diff
@@ -19,15 +19,34 @@ public class JsonEncoder
     @Getter
     private int offset = 0;
 
+    public int advanceOffset(final int diff)
+    {
+        return offset += diff;
+    }
+
     public void putDoubleAsString(final double value)
     {
         putQuote();
         putDouble(value);
         putQuote();
     }
 
-    public <E, A> void encodeArray(
-        final A array,
+    public void putByteBufferAsBase64(final ByteBuffer bufferToEncode)
+    {
+        putByteBufferAsBase64(bufferToEncode,
+            bufferToEncode.position(),
+            bufferToEncode.limit() - bufferToEncode.position());
+    }
+
+    public void putByteBufferAsBase64(final ByteBuffer bufferToEncode, final int startIdx, final int srcLen)
+    {
+        putQuote();
+        final var encodedLength = B64Encoder.encode(startIdx, srcLen, offset, bufferToEncode, buffer.byteBuffer());
```

Comment:
buffer.byteBuffer() may return null if buffer wraps byte[]. So it is safer to pass MutableAsciiBuffer to B64Encoder.encode as a destination. Or maybe move this method to JsonEncoder, call putByte and update offset in putByte insead of doing it manually.